api/internal/infrastructure/mongo/deployment.go

```
@@ -166,13 +210,76 @@
+		// Page-based pagination
+		skip := (pagination.Page.Page - 1) * pagination.Page.PageSize
+		limit := pagination.Page.PageSize
+
+		// Add sorting
+		var sort bson.D
+		if pagination.Page.OrderBy != nil {
+			direction := 1 // default ascending
+			if pagination.Page.OrderDir != nil && *pagination.Page.OrderDir == "DESC" {
+				direction = -1
+			}
+			sort = bson.D{{*pagination.Page.OrderBy, direction}}
```

golangci-lint (1.62.2)

composites: go.mongodb.org/mongo-driver/bson/primitive.E struct literal uses unkeyed fields

(govet)